hera_mc/weather.py

```diff
@@ -0,0 +1,233 @@
+# to data volume and reduction strategies and periods)
+# Also please add the sensor information to docs/weather_sensor_details.txt
+# You can get the needed information using the katportal example scripts at:
+#   https://github.com/ska-sa/katportalclient/tree/master/examples
+weather_sensor_dict = {'wind_speed': {'sensor_name': 'anc_mean_wind_speed',
+                                      'units': 'm/s',
+                                      'reduction': 'mean', 'period': 60,
+                                      'description': "Mean of  ['wind.wind-speed', 'weather.wind-speed'] in (600 * 1.0s) window (report period: 1s)"},
+                       'wind_gust': {'sensor_name': 'anc_gust_wind_speed',
+                                     'units': 'm/s',
+                                     'reduction': 'max', 'period': 60,
+                                     'description': "Max of  ['wind.wind-speed', 'weather.wind-speed'] in (3 * 1.0s) window (report period: 1s)"},
+                       'wind_direction': {'sensor_name': 'anc_wind_wind_direction',
+                                          'units': 'deg', 'reduction': None,
+                                          'description': "Wind direction angle (report period: 10s)"},
+                       'temperature': {'sensor_name': 'anc_weather_temperature',
+                                       'units': 'deg Celsius', 'reduction': None,
+                                       'description': "Air temperature (report period: 10s)"},
+                       'humidity': {'sensor_name': 'anc_weather_humidity',
+                                    'units': 'percent',
+                                    'reduction': 'mean', 'period': 60,
+                                    'description': "Air humidity (report period: 10s)"},
+                       'pressure': {'sensor_name': 'anc_weather_pressure', 'units': 'mbar',
+                                    'reduction': 'mean', 'period': 60,
+                                    'description': "Barometric pressure (report period: 10s)"},
+                       'rain': {'sensor_name': 'anc_weather_rain', 'units': 'mm',
+                                'reduction': 'sum', 'period': 60,
+                                'description': "Rainfall (report period: 10s)"}}
+
+
+def _reduce_time_vals(times, vals, period, strategy='decimate'):
+    if not isinstance(period, (int, np.int)):
+        raise ValueError('period must be an integer')
+
+    times_keep, inds = np.unique(np.floor(times / period) * period, return_index=True)
+    if times_keep[0] < np.min(times):
+        times_keep = times_keep[1:]
+        inds = inds[1:]
+
+    if strategy == 'decimate':
+        vals_keep = vals[inds]
+    elif strategy == 'max':
+        vals_keep = []
+        times_keep = times_keep[:-1]
+        for count in range(len(times_keep)):
+            vals_keep.append(np.max(vals[inds[count]:inds[count + 1]]))
+        vals_keep = np.array(vals_keep)
```

Comment:
I feel like these stanzas could be made a bit more compact using a list comprehension or something, but it might hurt readability for no real gain.

I would however prefer to write `range(len(inds) - 1)` rather than the current expression. I think that's clearer than lopping off the last element of `times_keep`, which AFAICT is just done to reduce the length count. And that way it is clearer that the `range` call is tuned to index into the `inds` variable.

Reply:
sure, I'll change it.